Source/DataSources/ModelVisualizer.js

```diff
@@ -370,8 +376,68 @@ ModelVisualizer.prototype.getBoundingSphere = function (entity, result) {
   }
 
   const hasHeightReference = model.heightReference !== HeightReference.NONE;
-  if (hasHeightReference && !defined(model._clampedModelMatrix)) {
-    return BoundingSphereState.PENDING;
+  if (hasHeightReference) {
+    const scene = this._scene;
+    const globe = scene.globe;
+    const ellipsoid = globe.ellipsoid;
+    const terrainProvider = globe.terrainProvider;
+
+    const modelMatrix = model.modelMatrix;
+    scratchPosition.x = modelMatrix[12];
+    scratchPosition.y = modelMatrix[13];
+    scratchPosition.z = modelMatrix[14];
+    const cartoPosition = Ellipsoid.WGS84.cartesianToCartographic(
+      scratchPosition
+    );
+
+    // Since the model's bounding sphere may be clamped to a lower LOD tile if
+    // the camera is initially far away, we use sampleTerrainMostDetailed to estimate
+    // where the bounding sphere should be and set that as the target bounding sphere
+    // for the camera.
+    let clampedBoundingSphere = this._modelHash[entity.id]
+      .clampedBoundingSphere;
+
+    if (!defined(clampedBoundingSphere)) {
+      clampedBoundingSphere = new BoundingSphere();
+
+      // Since this function is called per-frame, we set a flag when sampleTerrainMostDetailed
+      // is called and check for it to avoid calling it again.
+      const awaitingSampleTerrain = this._modelHash[entity.id]
+        .awaitingSampleTerrain;
+      if (!awaitingSampleTerrain) {
+        Cartographic.clone(cartoPosition, scratchCartographic);
+        sampleTerrainMostDetailed(terrainProvider, [scratchCartographic]).then(
+          () => {
+            // Update the height of the cartographic position of the model with the most detailed height.
+            if (model.heightReference === HeightReference.RELATIVE_TO_GROUND) {
+              scratchCartographic.height += cartoPosition.height;
+            }
+            ellipsoid.cartographicToCartesian(
+              scratchCartographic,
+              scratchPosition
+            );
+
+            // Update the bounding sphere with the updated position.
+            BoundingSphere.clone(model.boundingSphere, clampedBoundingSphere);
+            clampedBoundingSphere.center = scratchPosition;
+
+            this._modelHash[
+              entity.id
+            ].clampedBoundingSphere = BoundingSphere.clone(
+              clampedBoundingSphere
+            );
+          }
+        );
+        this._modelHash[entity.id].awaitingSampleTerrain = true;
```

Comment:
- Expected this get set to `false` when the promise is resolved, but didn't see that.
- From a readabilty perspective it would be better to set this before the then clause.  That puts key change all at the top of the if.  It also keeps the code in the same reading order as it would be executed.  Now I'm reading through the `then` where I have the terrain and get to this line and have to rewind back in time to before the `then` ran.  This is also why `async/await` tends to read better.

Reply:
> Expected this get set to false when the promise is resolved, but didn't see that.

Good catch. Fixed.

> From a readabilty perspective it would be better to set this before the then clause. 

I agree, updated.